lib/dns.js

```diff
@@ -106,6 +106,9 @@ exports.lookup = function lookup(hostname, options, callback) {
         hints !== (exports.ADDRCONFIG | exports.V4MAPPED)) {
       throw new TypeError('invalid argument: hints must use valid flags');
     }
+
+    if (process.platform === 'freebsd' || family !== 6)
```

Comment:
Oh, and I might have pasted the wrong code on IRC; I think that should be `&&`, not `||`.  Mea culpa!